mapdrawgdal.c

```diff
@@ -1937,9 +1995,8 @@ msDrawRasterLayerGDAL_16BitClassification(
       /*
        * Skip nodata pixels ... no processing.
        */
-      if( bGotNoData && fRawValue == fNoDataValue ) {
+      if( bGotNoData && IsNoData(fRawValue, fNoDataValue, bIsNoDataNan) )
         continue;
-      }
 
       if( CPLIsNan(fRawValue) )
```

Comment:
Yes, CPLIsNan is already there. IsNoData is still required to check if fRawValue is No Data when NoData is not NaN. Then CPLIsNaN is redundant.